### ISSUE TITLE
support cpu limit in docker

### ISSUE THREAD

**Issue by okhowang** [ninja-build/ninja#1824]
docker can limit containers cpu usage by `docker run --cpus {num}`.
ninja 1.10 couldn't read it currently.
so if I run `docker run --cpus 2 some_image ninja`
it will run 10 sub process (my cpu is 8 core).
```
[okhowang@VM_51_247_centos /home/linuxbrew/.linuxbrew]$ docker run -it --rm --cpus 3 -v$PWD:$PWD -w$PWD ubuntu:20.04 bin/ninja --version
1.10.0
[okhowang@VM_51_247_centos /home/linuxbrew/.linuxbrew]$ docker run -it --rm --cpus 3 -v$PWD:$PWD -w$PWD ubuntu:20.04 bin/ninja --help
usage: ninja [options] [targets...]

if targets are unspecified, builds the 'default' target (see manual).

options:
  --version      print ninja version ("1.10.0")
  -v, --verbose  show all command lines while building

  -C DIR   change to DIR before doing anything else
  -f FILE  specify input build file [default=build.ninja]

  -j N     run N jobs in parallel (0 means infinity) [default=10 on this system]
  -k N     keep going until N jobs fail (0 means infinity) [default=1]
  -l N     do not start new jobs if the load average is greater than N
  -n       dry run (don't run commands but act like they succeeded)

  -d MODE  enable debugging (use '-d list' to list modes)
  -t TOOL  run a subtool (use '-t list' to list subtools)
    terminates toplevel options; further flags are passed to the tool
  -w FLAG  adjust warnings (use '-w list' to list warnings)
```
windows container has some problem when isolation is process
```
C:\ProgramData\chocolatey\lib\ninja\tools> docker run -it --rm --cpus 4 --isolation=process -v "${PWD}:$PWD" -w $PWD mcr.microsoft.com/windows/nanoserver:2004 ninja --help
usage: ninja [options] [targets...]

if targets are unspecified, builds the 'default' target (see manual).

options:
  --version      print ninja version ("1.10.0")
  -v, --verbose  show all command lines while building

  -C DIR   change to DIR before doing anything else
  -f FILE  specify input build file [default=build.ninja]

  -j N     run N jobs in parallel (0 means infinity) [default=10 on this system]
  -k N     keep going until N jobs fail (0 means infinity) [default=1]
  -l N     do not start new jobs if the load average is greater than N
  -n       dry run (don't run commands but act like they succeeded)

  -d MODE  enable debugging (use '-d list' to list modes)
  -t TOOL  run a subtool (use '-t list' to list subtools)
    terminates toplevel options; further flags are passed to the tool
  -w FLAG  adjust warnings (use '-w list' to list warnings)
C:\ProgramData\chocolatey\lib\ninja\tools> docker run -it --rm --cpus 4 --isolation=process -v "${PWD}:$PWD" -w $PWD mcr.microsoft.com/windows/nanoserver:2004 ninja --version
1.10.0
```

**Comment by jhasse** [comment]
Thanks for the report! This should have been fixed by 1bcc689324bdee090eed035353724abc3fa7c909, but apparently not. Can you check if it works if you compile master from source? Maybe `CPU_COUNT` wasn't set or the `sched_getaffinity` method doesn't work after all.

**Comment by okhowang** [comment]
I have built ninja from source with master, and got same result.
